packages/trpc/server/routers/viewer/teams/removeMember.handler.ts

```diff
@@ -54,6 +55,38 @@ export const removeMemberHandler = async ({ ctx, input }: RemoveMemberOptions) =
 
   if (input.isOrg) {
     // Deleting membership from all child teams
+    const foundUser = await ctx.prisma.user.findUnique({
+      where: { id: input.memberId },
+      select: {
+        email: true,
+      },
+    });
+
+    const orgInfo = await ctx.prisma.team.findUnique({
+      where: { id: input.teamId },
+      select: {
+        metadata: true,
+      },
+    });
+
+    if (!foundUser || !orgInfo) throw new TRPCError({ code: "NOT_FOUND" });
+
+    const parsedMetadata = teamMetadataSchema.parse(orgInfo.metadata);
+
+    if (
+      parsedMetadata?.isOrganization &&
+      parsedMetadata.isOrganizationVerified &&
+      parsedMetadata.orgAutoAcceptEmail
+    ) {
+      if (foundUser.email.endsWith(parsedMetadata.orgAutoAcceptEmail)) {
+        await ctx.prisma.user.delete({
+          where: { id: input.memberId },
+        });
+        // This should cascade delete all memberships and hosts etc
+        return;
+      }
+    }
```

Comment:
I put these into seperate IFs so its more read-able.

If the team is an organization & is verified & there is a autoAcceptDomain
AND
if the found users email ends with the autoAcceptDomain
Delete the user